multi_import/data.py

```diff
@@ -227,5 +239,32 @@ def get_http_response(self, file_format=None, filename=None):
         response["Content-Disposition"] = header
         return response
 
+    def _write_tree_export(self, format, result, file: zipfile.ZipFile) -> None:
+        """
+        Write exported content items in individual files with the items'
+        id_column value as the filenames.
+        """
+        dataset = result.get_dataset(format)
+
+        for row in dataset:
+            new_ds = Dataset(headers=dataset.headers)
+            new_ds.append(row)
+            file_contents = format.write(new_ds)
+            item_id = new_ds.dict[0][result.id_column]
+            file_name = f"{item_id}.{format.extension}"
+            file.writestr(f"{result.filename}/{file_name}", file_contents.getvalue())
+
+    def _write_tabular_export(self, format, result, file: zipfile.ZipFile) -> None:
+        """
+        Write exported content items in the same file with the content item
+        type as the filename.
+        """
+        file = result.get_file(format)
+        fname = "{0}.{1}".format(result.filename, format.extension)
+        file.writestr(fname, file.getvalue())
+
     def _get_format(self, file_format):
         return files.find_format(self.file_formats, file_format)
+
+    def _is_template_export(self):
+        return len(self.results) == 1
```

Comment:
This check is for the individual content library pages, such as the library tasks list page... because the user is downloading a single content type (such as tasks), we don't bother zipping the output.

`get_http_response` does result in `get_file` being called, but also that the correct HTTP headers are set for the single file.

Looks like we are doing a very similar check, in `get_file`... so could update this line to use the `_is_single_content_type_export` method that you just added.